android/src/main/java/io/wifi/p2p/MessageServer.java

```diff
@@ -0,0 +1,107 @@
+package io.wifi.p2p;
+
+import static io.wifi.p2p.Utils.CHARSET;
+
+import android.os.Bundle;
+import android.util.Log;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * Created by kiryl on 18.7.18. 
+ * Refactor by viniciuscb on 20.03.23 
+ *
+ * A simple server socket that accepts connection and writes
+ * some data on the stream.
+ */
+public class MessageServer {
+  private static final String TAG = "RNWiFiP2P";
+  private Callback callback;
+  private final Executor executor;
+  private volatile ServerSocket serverSocket;
+  private volatile boolean returnMeta = false;
+
+  public MessageServer() {
+    this(Executors.newSingleThreadExecutor());
+  }
+
+  public MessageServer(Executor executor) {
+    this.executor = executor;
+  }
+
+  public void start(ReadableMap props, Callback callback) {
+    executor.execute(
+        () -> {
+          try {
+            serverSocket = new ServerSocket(8988);
+            Log.i(TAG, "Server: Socket opened");
+            this.callback = callback;
+
+            if (props != null) {
+              Bundle bundle = Arguments.toBundle(props);
+              this.returnMeta = bundle.getBoolean("meta");
+            } else {
+              this.returnMeta = false;
+            }
+
+            while (!Thread.currentThread().isInterrupted()) {
```

Comment:
How much times this cycle is executed?

Maybe more convenient would to move `this.stop` out of the loop? Otherwise it seems like every iteration we are trying to close a connection, but in fact `stop` method does nothing in this case 🤔 

Could you please send a link to documentation why do we need to use `while (!Thread.currentThread().isInterrupted())` inside of `execute` method? I don't understand this piece of the code...

Reply:
I am reviewing this. Testing...